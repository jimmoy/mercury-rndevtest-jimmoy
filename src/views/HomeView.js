// HomeView.js

import React, { Component } from 'react'

import {
  Text,
  StyleSheet,
  TouchableOpacity,
  View
} from 'react-native'

const styles = StyleSheet.create({
  root: {
    flex: 1,
    flexDirection: 'column',
    marginTop: 20,
    justifyContent: 'flex-start',
    alignItems: 'center',
  },
  peopleContainer: {
    justifyContent: 'flex-start',
    alignItems: 'flex-start',
  },
  personContainerEven: {
    backgroundColor: '#dddddd',
  },
  personContainerOdd: {
    backgroundColor: '#ffffff',
  },
  person: {
    fontSize: 18,
  },
  button: {
    fontSize: 16,
    textAlign: 'center',
    padding: 10,
  },
})

export const OnePerson = ({ person }) =>
  <View style={styles.personContainer}>
    <Text style={styles.person}>{person.name}</Text>
  </View>

const rowStyleSuffix = n => (n % 2 === 0) ? 'Even' : 'Odd'

export const PeopleList = ({ people }) =>
  people.map((person, index) => {
    const styleName = `personContainer${rowStyleSuffix(index)}`
    return (
      <View key={person.name} style={styles[styleName]}>
        <Text style={styles.person}>{person.name}</Text>
      </View>
      )
  })

export const HomeView  = ({ data, onClick }) =>
  <View style={styles.root}>
    <View style={styles.peopleContainer}>
      <PeopleList people={data} />
    </View>
    <TouchableOpacity onPress={onClick}>
      <Text style={styles.button}>Button</Text>
    </TouchableOpacity>
  </View>
