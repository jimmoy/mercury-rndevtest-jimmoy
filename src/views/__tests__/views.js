import React from 'react'
import renderer from 'react-test-renderer'

import { HomeView } from '../HomeView'

describe('snapshot tests', () => {
  test('HomeView', () => {
    const rendered = renderer.create(<HomeView data={[]} />)
    expect(rendered.toJSON()).toMatchSnapshot()
  })
})
