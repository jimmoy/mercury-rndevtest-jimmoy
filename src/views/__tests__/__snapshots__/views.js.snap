// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot tests HomeView 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "marginTop": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 300,
        "justifyContent": "center",
        "width": 300,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 18,
        }
      }
    >
      Network request...
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "padding": 10,
          "textAlign": "center",
        }
      }
    >
      Button
    </Text>
  </View>
</View>
`;
